StitchCore/Tests/StitchCoreTests/Services/Internal/CoreStitchServiceTests.swift

```diff
@@ -18,11 +18,11 @@ class CoreStitchServiceTests: XCTestCase {
             return Response.init(statusCode: 200, headers: [:], body: nil)
         }
 
-        func doAuthenticatedJSONRequest(_ stitchReq: StitchAuthDocRequest) throws -> Any {
+        func doAuthenticatedJSONRequest<T: Decodable>(_ stitchReq: StitchAuthDocRequest) throws -> T {
             XCTAssertEqual(stitchReq.method, .post)
             XCTAssertEqual(stitchReq.path, appRoutes.serviceRoutes.functionCallRoute)
             XCTAssertEqual(stitchReq.document, expectedDoc)
-            return self
+            return try JSONDecoder().decode(T.self, from: Data())
```

Comment:
[nit] `BsonDecoder`?

Reply:
Actually this may need to be decoded into a document. BsonDecoder is for Bson, not extended json.

Reply:
Can just use `try BsonDecoder().decode(T.self, from: responseString)`. That's how you're doing it in the request client.

Reply:
A test that has something like `{"a": {"$numberDouble": "52.43"}}` would be nice to see that it decodes to `{"a": 52.43}`